src/shared/clients/s3Client.ts

```diff
@@ -303,7 +303,7 @@ export class DefaultS3Client {
      *
      * @throws Error if there is an error calling S3 or piping between streams.
      */
-    public async uploadFile(request: UploadFileRequest): Promise<void> {
+    public async uploadFile(request: UploadFileRequest): Promise<any> {
```

Comment:
We've just introduce a bug in the code if you didn't update _all_ callers of `uploadFile` to await the promise returned by `managedUpload`. Before this function would not return until the upload is finished, while now it returns almost immediately.